support/interactive/interactive.cc

```diff
@@ -15,56 +15,60 @@
 
 int report_fd, walltimelimit;
 
-int val_pid = -1, user_pid = -1;
-int user_status = -1, val_status = -1;
-static rusage user_ru;
-static rusage val_ru;
+volatile bool validator_first = false;
+volatile int val_pid = -1, user_pid = -1;
+volatile int user_status = -1, val_status = -1;
+volatile static rusage user_ru;
+volatile static rusage val_ru;
```

Comment:
Why do you think these (and others below) need to be volatile?

Reply:
They are used from a signal handler, so it seemed "more correct" (even if still actually incorrect -- only `volatile sig_atomic_t`s and atomics are allowed, and even then we have race conditions that make the whole approach wrong). But until the signal handler is fixed properly, this may at least prevent the compiler from reordering some assignments etc. which I wouldn't be surprised if it does.

Reply:
Makes sense!